test/server/controller.spec.js

```diff
@@ -219,224 +232,343 @@ describe('Handling requests - error responses', function() {
+          expect(JSON.parse(body)).toEqual({
+            htsget:{
+              error:   "NotFound",
+              message: "URL not found : /invalid"
+            }
+          });
+          done();
         });
-        done();
       });
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam"}), () => {
+          req.end();
+        });
+      } else {
+        req.end();
+      }
     });
-    req.end();
-  });
 
-  it('Invalid input error for duplicate values for an attribute', ( done ) => {
-    server.removeAllListeners('request');
-    server.on('request', (request, response) => {
-      let c = new RangerController(request, response, {one: "two"});
-      expect( () => {c.handleRequest();} ).not.toThrow();
+    it(`Not found error, auth checked for ${obj.method}`, function(done) {
+      server.removeAllListeners('request');
+      server.on('request', (request, response) => {
+        let c = new RangerController(request, response, {one: "two"});
+        expect( () => {c.handleRequest();} ).not.toThrow();
+      });
+      obj.path = '/invalid';
+      let req = http.request( obj );
+      req.setHeader('X-Remote-User', 'user1');
+      req.on('response', function(response) {
+        var body = '';
+        response.on('data', function(d) { body += d;});
+        response.on('end', function() {
+          checkGenericErrorResponse(expect, response);
+          expect(response.statusCode).toEqual(404);
+          expect(response.statusMessage).toEqual('URL not found : /invalid');
+          expect(JSON.parse(body)).toEqual({
+            htsget: {
+              error:   "NotFound",
+              message: "URL not found : /invalid"
+            }
+          });
+          done();
+        });
+      });
+      if (obj.method === "POST") {
+        req.write(JSON.stringify({"format":"bam"}), () => {
+          req.end();
+        });
+      } else {
+        req.end();
+      }
     });
 
-    http.get({socketPath: socket, path: '/sample?attr1=value1&attr2=value2&attr1=value3'},
-      ( response ) => {
-      var body = '';
-      response.on('data', ( d ) => { body += d;});
-      response.on('end', () => {
-        checkGenericErrorResponse(expect, response);
-        expect(response.statusCode).toEqual(400);
-        let m = "Invalid request: multiple values for attribute 'attr1'";
-        expect(response.statusMessage).toEqual(m);
-        expect(JSON.parse(body)).toEqual({
-          htsget: {
-            error:   ServerHttpError.INVALID_INPUT,
-            message: m
-          }
+    it(`Invalid input error for duplicate values for an attribute for ${obj.method}`, ( done ) => {
+      server.removeAllListeners('request');
+      server.on('request', (request, response) => {
+        let c = new RangerController(request, response, {one: "two"});
+        expect( () => {c.handleRequest();} ).not.toThrow();
+      });
+      obj.path = '/sample?attr1=value1&attr2=value2&attr1=value3';
```

Comment:
obj.path needs to be specific to the kind of request:
```
if (obj.method === "POST") {
  obj.path = '/sample';
} else {
  obj.path = '/sample?attr1=value1&attr2=value2&attr1=value3';
}
```
Should not change anything but the test will be more explicit.